packages/reporters/liquidation-reporter/liquidation-reporter.ts

```diff
@@ -0,0 +1,269 @@
+        (info: any) => toChecksumAddress(info.id) === toChecksumAddress(financialContractAddress)
+      )[0];
+      // If there is no position info, then continue. This would happen if there is data that is not on the graph.
+      if (contractPositionsInfo == undefined) {
+        continue;
+      }
+
+      // If the contract has no positions, we should also continue. Remove the contract from the list of active contracts
+      if (contractPositionsInfo.positions.length == 0) {
+        uniqueCollateralTypes[collateralAddress].activeFinancialContracts = removeItemFromArrayOfObjects(
+          uniqueCollateralTypes[collateralAddress].activeFinancialContracts,
+          "contractAddress",
+          financialContractAddress
+        );
+        continue;
+      }
+      // If we got to this point we know that the financial contract matches what was returned from the graph and
+      // the contract has positions in it. We can now fetch some extra info about the contract.
+      const financialContract = new web3.eth.Contract(getAbi("ExpiringMultiParty"), financialContractAddress);
+      const [collateralRequirementString, contractExpirationTime, contractPriceIdentifier] = await Promise.all([
+        financialContract.methods.collateralRequirement().call(),
+        financialContract.methods.expirationTimestamp().call(),
+        financialContract.methods.priceIdentifier().call()
+      ]);
+      const collateralRequirement = toBN(collateralRequirementString);
+
+      // If expired, remove it from the list as it is not active.
+      if (getTime() > Number(contractExpirationTime)) {
+        uniqueCollateralTypes[collateralAddress].activeFinancialContracts = removeItemFromArrayOfObjects(
+          uniqueCollateralTypes[collateralAddress].activeFinancialContracts,
+          "contractAddress",
+          financialContractAddress
+        );
+        continue;
+      }
+      uniqueCollateralTypes[collateralAddress].activeFinancialContracts[financialContractIndex] = {
+        ...uniqueCollateralTypes[collateralAddress].activeFinancialContracts[financialContractIndex],
+        contractPriceIdentifier: hexToUtf8(contractPriceIdentifier),
+        collateralRequirement: Number(fromWei(collateralRequirement)),
+        contractExpirationTime
+      };
+
+      // Else, we can start building up draw down information.
+      try {
+        // Create a price feed for the product. We need to know the identifier price feed to compute drawdown as this
+        // considers either the collateral value dropping OR the synthetic value increasing. We can set all three
+        // common API keys. If the feed does not need one of these it just ignores it.
+        const samplePriceFeed = await createReferencePriceFeedForFinancialContract(
+          logger,
+          web3,
+          networker,
+          getTime,
+          financialContractAddress,
+          {
+            cryptowatchApiKey: process.env.CRYPTO_WATCH_API_KEY,
```

Comment:
do you not need to import `require('dotenv').config()` to use this?

Reply:
dont think so? it seemed to work fine. I think that's if you want to read in a `.env` file. if you want to read directly from the `export ENV=xxx` then it's ok. this will be the case in production.